File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/MethodType.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

/**
 * Method with which the end user is redirected to the Instore Selfservice application.
 */
public enum MethodType {

    SMS("SMS"),
    URL("URL");

    private String value;

    MethodType(String value) {
        this.value = value;
    }

    public static MethodType fromValue(String value) {
        for (MethodType b : MethodType.values()) {
            if (b.value.equals(value)) {
                return b;
            }
        }
        throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

File: upl-purchase-sdk-js/src/unzer-paylater/model/MethodType.js
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

/**
 * Method with which the end user is redirected to the Instore Selfservice application.
 */

const ValidationException = require("../exceptions/ValidationException");

module.exports = Object.freeze({
    SMS: "SMS",
    URL: "URL",
    constructFromObject: function (data) {
        if (data === null || data === undefined || this.hasOwnProperty(data)) {
            return data;
        }
        throw new ValidationException(`Could not find '${data}' in String.`);
    },
});
